### ISSUE TITLE
Update contributing.md docs review guidelines

### ISSUE THREAD

**Issue by zou3519** [pytorch/pytorch#55790]
## 📚 Documentation

I noticed recently that circleci basically gives the docs build as an artifact if it succeeds (see [here for an example](https://app.circleci.com/pipelines/github/pytorch/pytorch/298742/workflows/c72e6b5b-dc64-4a3e-9120-03bc75219d36/jobs/12258611/artifacts)). We should update [our guidelines](https://github.com/pytorch/pytorch/blob/master/CONTRIBUTING.md#submitting-changes-for-review) for submitting docs changes for review; previously we would ask users to build docs changes locally and send us screenshots but now we can inspect them directly before they go in

cc @brianjo @mruberry

**Comment by kshitij12345** [comment]
Wiki Note does mention about that (but agreed it would be more accessible if it is on the contributing.md)
https://github.com/pytorch/pytorch/wiki/Developer-FAQ#if-im-editing-the-documentation-how-do-i-see-the-formatted-doc

**Comment by brianjo** [comment]
Yes, the artifacts are in the doc build. I was wonder if we could do a step in GitHub that added a link to the index.html file.
